### PR TITLE
ISIN mapping endpoint was added.

### PR DESCRIPTION
**Pull request by bguban**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ### 1.1.2 (Next)
 
 * Your contribution here.
+* [#64](https://github.com/dblock/iex-ruby-client/pull/64): ISIN mapping endpoint was added - [@bguban](https://github.com/bguban).
```
Comment (dblock) [inline CHANGELOG.md]: Or with fewer words, "Add `isin_mapping`" :)

File: README.md [hunk 2]
```diff
@@ -372,6 +373,27 @@ crypto.high_dollar #'$3,590'
...
+```ruby
+symbols = client.isin_mapping(['US0378331005'])
+
+symbols.first.exchange # NAS
+symbols.first.iex_id # IEX_4D48333344362D52
+symbols.first.region # US
+symbols.first.symbol # AAPL
+```
+
+You also can use `mapped: true` option to receive symbols mapped by their ISINs.
```
Comment (dblock) [inline README.md]: You can use ... (no need for also).

File: lib/iex/endpoints/ref_data.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+module IEX
+  module Endpoints
+    module RefData
+      def isin_mapping(isins, options = {})
```
Comment (dblock) [inline lib/iex/endpoints/ref_data.rb]: The API is `asin`, why do we call it `asin_mapping`? Shouldn't it be `asin`?

File: lib/iex/endpoints/ref_data.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        response = post('ref-data/isin', { token: publishable_token, isin: isins }.merge(options))
+
+        if response.is_a?(Hash) # mapped option was set
```
Comment (dblock) [inline lib/iex/endpoints/ref_data.rb]: I really dislike these APIs that return two different things depending on what you pass into them. Why don't we make two methods: `isin` and `mapped_isin`?

File: lib/iex/resources.rb [hunk 5]
```diff
@@ -13,3 +13,4 @@
 require_relative 'resources/quote'
 require_relative 'resources/sectors'
 require_relative 'resources/crypto'
+require_relative 'resources/isin_mapping'
```
Comment (dblock) [inline lib/iex/resources.rb]: `resources/isin` to be consistent with the API

Comment (bguban) [reply lib/iex/resources.rb]: `Helper call to convert ISIN to IEX Cloud symbols` from https://iexcloud.io/docs/api/#isin-mapping. Maybe it's better to call the resource `symbol`? WDYT?

File: lib/iex/endpoints/ref_data.rb (hunk 3 above)
Comment (bguban) [reply lib/iex/endpoints/ref_data.rb]: I called it as the documentation section was called (https://iexcloud.io/docs/api/#isin-mapping). As for me, it's a very controversial endpoint. It even doesn't return ISINs. It returns symbols that mutch to given ISINs.
Maybe it even would be better to call it `ref_data_isin`. What do you think?

File: lib/iex/resources.rb (hunk 5 above)
Comment (dblock) [reply lib/iex/resources.rb]: I worry about something so generic as `symbol`, but I'm OK with it. Your call, it does return a symbol with its metadata.

Generally the client should not be too opinionated. If IEX calls something a certain way we should stick to it.

Comment (bguban) [reply lib/iex/resources.rb]: In this case, should we have two endpoints like `isin` and `isin_mapped` if IEX returns different responses depending on the request options?

File: lib/iex/endpoints/ref_data.rb (hunk 4 above)
Comment (bguban) [reply lib/iex/endpoints/ref_data.rb]: As you mentioned in another comment the client should not be too opinionated. Keeping it in mind, I think it's better to copy the behavior of IEX endpoint and have two different responses depending on options in the single method. 
Again, if split the endpoint to two methods, the end-user can call `isin` endpoint passing there  `mapped: true` option what causes an error while the response parsing. So it requires an end-user to know the client-specific method `mapped_isin`.

File: spec/iex/endpoints/ref_data_spec.rb [hunk 6]
```diff
@@ -3,14 +3,28 @@
...
+
+    it 'converts ISIN to IEX Cloud symbols' do
       expect(subject.count).to eq(2)
-      expect(subject.first).to eq('exchange' => 'NAS', 'iexId' => 'IEX_4D48333344362D52', 'region' => 'US', 'symbol' => 'AAPL')
-      expect(subject.last).to eq('exchange' => 'ETR', 'iexId' => 'IEX_464D46474C312D52', 'region' => 'DE', 'symbol' => 'APC-GY')
+      expect(subject.first).to eq('exchange' => 'NAS', 'iex_id' => 'IEX_4D48333344362D52', 'region' => 'US', 'symbol' => 'AAPL')
+      expect(subject.last).to eq('exchange' => 'ETR', 'iex_id' => 'IEX_464D46474C312D52', 'region' => 'DE', 'symbol' => 'APC-GY')
+    end
+
+    context 'with mapped option', vcr: { cassette_name: 'ref-data/isin_mapped' } do
```
Comment (dblock) [inline spec/iex/endpoints/ref_data_spec.rb]: This context does not depend/inherit the parent one. Split them up here so each has its own section.

**Comment by dblock** [comment]
LGTM after the spec block change, feel free to merge

**Pull request merged**